ntbs-service/Pages/Search/Index.cshtml

```diff
@@ -26,43 +27,44 @@
         <p> To filter results, add more search terms and search again. </p>
     </div>
-    <div>
+    <ol class="search-results-list">
         @foreach(var result in @Model.SearchResultsBannerDisplay) {
-            <partial name="../Shared/_NotificationBanner.cshtml" model="result"/>
+            <li>
+                <partial name="../Shared/_NotificationBanner.cshtml" model="result"/>
+            </li>
         }
-    </div>
+    <ol>
 
     <nav class="nhsuk-pagination" role="navigation" aria-label="Pagination">
         <ul class="nhsuk-list nhsuk-pagination__list">
-            @if(@Model.Notifications.HasPreviousPage)
-            {
-                <li class="nhsuk-pagination-item--previous">
-                    <a class="nhsuk-pagination__link nhsuk-pagination__link--prev" asp-page="./Index" asp-route-pageIndex="@(Model.PageIndex - 1)">
-                        <span class="nhsuk-pagination__title">Previous</span>
-                        <span class="nhsuk-u-visually-hidden">:</span>
-                        <span class="nhsuk-pagination__page"> Page @(@Model.PageIndex - 1) of @Model.Notifications.TotalPages </span>
-                        <svg class="nhsuk-icon nhsuk-icon__arrow-left" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" aria-hidden="true">
-                        <path d="M4.1 12.3l2.7 3c.2.2.5.2.7 0 .1-.1.1-.2.1-.3v-2h11c.6 0 1-.4 1-1s-.4-1-1-1h-11V9c0-.2-.1-.4-.3-.5h-.2c-.1 0-.3.1-.4.2l-2.7 3c0 .2 0 .4.1.6z"></path>
-                        </svg>
-                    </a>
-                </li>
-            }
-            @if(@Model.Notifications.HasNextPage)
-            {
-                <li class="nhsuk-pagination-item--next">
-                    <a class="nhsuk-pagination__link nhsuk-pagination__link--next" asp-page="./Index" asp-route-pageIndex="@(Model.PageIndex + 1)">
-                        <span class="nhsuk-pagination__title">Next</span>
-                        <span class="nhsuk-u-visually-hidden">:</span>
-                        <span class="nhsuk-pagination__page"> Page @(@Model.PageIndex + 1) of @Model.Notifications.TotalPages </span>
-                        <svg class="nhsuk-icon nhsuk-icon__arrow-right" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" aria-hidden="true">
-                        <path d="M19.6 11.66l-2.73-3A.51.51 0 0 0 16 9v2H5a1 1 0 0 0 0 2h11v2a.5.5 0 0 0 .32.46.39.39 0 0 0 .18 0 .52.52 0 0 0 .37-.16l2.73-3a.5.5 0 0 0 0-.64z"></path>
-                        </svg>
-                    </a>
-                </li>
+            @{
+
+                var previousPageText = "";
+                var previousPageUrl = "";
+                var nextPageText = "";
+                var nextPageUrl = "";
+                var queryString = Request.Query;
+                var previousPageQueryString = new RouteValueDictionary();
+                queryString.Keys.Select(x => previousPageQueryString[x] = queryString[x].ToString());

```

Comment:
Ah, this is a very confusing way of using `Select`, since you're not just transforming the values, but actually assinging them, which is much more like a `for each` loop/method.
Perhaps parsing and building them using these helpers could be helpful? https://www.talkingdotnet.com/build-query-string-asp-net-core/

And yes, I think we should move this logic to the controller and expose a set of properties on the model that this view can consume (through e.g. @Model.NextPageText, @Model.NextPageUrl).